lib/mauth/client/security_token_cacher.rb

```diff
@@ -54,12 +48,17 @@ def security_token_from(response_body)
         def signed_mauth_connection
           require 'faraday'
           require 'mauth/faraday'
-          @mauth_client.faraday_options[:ssl] = { ca_path: @mauth_client.ssl_certs_path } if @mauth_client.ssl_certs_path
-          @signed_mauth_connection ||= ::Faraday.new(@mauth_client.mauth_baseurl, @mauth_client.faraday_options) do |builder|
-            builder.use MAuth::Faraday::MAuthClientUserAgent
-            builder.use MAuth::Faraday::RequestSigner, 'mauth_client' => @mauth_client
-            builder.use :http_cache, serializer: Oj, logger: MAuth::Client.new.logger, shared_cache: false
-            builder.adapter ::Faraday.default_adapter
+          @signed_mauth_connection ||= begin
+            if @mauth_client.ssl_certs_path
+              @mauth_client.faraday_options[:ssl] = { ca_path: @mauth_client.ssl_certs_path }
+            end
+
+            ::Faraday.new(@mauth_client.mauth_baseurl, @mauth_client.faraday_options) do |builder|
+              builder.use MAuth::Faraday::MAuthClientUserAgent
+              builder.use MAuth::Faraday::RequestSigner, 'mauth_client' => @mauth_client
+              builder.use :http_cache, logger: MAuth::Client.new.logger, shared_cache: false
```

Comment:
Using `serializer: Oj` in :http_cache causes a mysterious exception...
```
     NoMethodError:
       undefined method `clear' for nil:NilClass
     # .../gems/2.6.0/gems/faraday-1.7.2/lib/faraday/utils/headers.rb:102:in `replace'
     # .../gems/2.6.0/gems/faraday-http-cache-2.2.0/lib/faraday/http_cache/storage.rb:142:in `load'
```

Removed to use the default serializer `JSON`